lib/lsp-data/z3950_connection.rb

```diff
@@ -23,4 +24,23 @@ def initialize(host:, database_name:, port: 210, element_set_name: 'F', credenti
       @connection = conn
     end
   end
+
+  ### index is a Bib-1 Use Attribute (see https://www.loc.gov/z3950/agency/defns/bib1.html)
+  ### nil records are included in results array to allow further action if needed
+  def search(index:, identifier:)
+    search_string = "@attr 1=#{index} #{identifier}"
+    response = connection.search(search_string)
+    results = []
+    response.records.each do |result|
+      results << (result.nil? ? result : record_from_result(result))
+    end
+    results
```

Comment:
Does this have the same effect?
```suggestion
    response.records.map do |result|
      result.nil? ? result : record_from_result(result)
    end
```